clients/spring-boot-starter-camunda-sdk/src/main/java/io/camunda/spring/client/jobhandling/DefaultJobExceptionHandlingStrategy.java

```diff
@@ -0,0 +1,115 @@
+  private final MetricsRecorder metricsRecorder;
+
+  public DefaultJobExceptionHandlingStrategy(
+      final CommandExceptionHandlingStrategy commandExceptionHandlingStrategy,
+      final MetricsRecorder metricsRecorder) {
+    this.commandExceptionHandlingStrategy = commandExceptionHandlingStrategy;
+    this.metricsRecorder = metricsRecorder;
+  }
+
+  private CommandWrapper createCommandWrapper(
+      final FinalCommandStep<?> command, final ActivatedJob job, final JobWorkerValue workerValue) {
+    return new CommandWrapper(
+        command,
+        job,
+        commandExceptionHandlingStrategy,
+        metricsRecorder,
+        workerValue.getMaxRetries());
+  }
+
+  @Override
+  public void handleException(final Exception exception, final ExceptionHandlingContext context)
+      throws Exception {
+    if (exception instanceof final JobError jobError) {
+      LOG.trace("Caught job error on {}", context.job());
+      final CommandWrapper command =
+          createCommandWrapper(
+              createFailJobCommand(context.jobClient(), context.job(), jobError),
+              context.job(),
+              context.jobWorkerValue());
+      command.executeAsyncWithMetrics(
+          MetricsRecorder.METRIC_NAME_JOB, MetricsRecorder.ACTION_FAILED, context.job().getType());
+    } else if (exception instanceof final BpmnError bpmnError) {
+      LOG.trace("Caught BPMN error on {}", context.job());
+      final CommandWrapper command =
+          createCommandWrapper(
+              createThrowErrorCommand(context.jobClient(), context.job(), bpmnError),
+              context.job(),
+              context.jobWorkerValue());
+      command.executeAsyncWithMetrics(
+          MetricsRecorder.METRIC_NAME_JOB,
+          MetricsRecorder.ACTION_BPMN_ERROR,
+          context.job().getType());
+    } else {
+      metricsRecorder.increase(
+          MetricsRecorder.METRIC_NAME_JOB, MetricsRecorder.ACTION_FAILED, context.job().getType());
+      throw exception;
+    }
+  }
+
+  private FinalCommandStep<Void> createThrowErrorCommand(
+      final JobClient jobClient, final ActivatedJob job, final BpmnError bpmnError) {
+    final ThrowErrorCommandStep2 command =
+        jobClient
+            .newThrowErrorCommand(job.getKey())
+            .errorCode(bpmnError.getErrorCode())
+            .errorMessage(bpmnError.getErrorMessage());
+    return JobHandlingUtil.applyVariables(bpmnError.getVariables(), command);
+  }
+
+  private FinalCommandStep<FailJobResponse> createFailJobCommand(
+      final JobClient jobClient, final ActivatedJob job, final JobError jobError) {
+    final int retries =
+        jobError.getRetries() == null ? (job.getRetries() - 1) : jobError.getRetries();
+    final StringWriter stringWriter = new StringWriter();
+    final PrintWriter printWriter = new PrintWriter(stringWriter);
+    jobError.printStackTrace(printWriter);
+    final String message = stringWriter.toString();
```

Comment:
🔧 just noticed `jobError.errorMessage` is not used here, would be worth to prepend, wdyt?

Reply:
ah I overlooked that it is passed along 👍

Reply:
An added test asserts that the submitted message is part of the error message. The error message is created from a separate util function, so the test is consistent